Golang segment unexpected rendering behaviour

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

I had the golang segment with full version setup in a tooltip - see theme file attached. The segment seems to cause rendering issues with text at the cursor. As I type (iTerm on M1 Mac) spaces are added and removed making the text input seem garbled. The tool tip also seems to render one character in from the right margin. The segment created odd vertical position re-rendering (going up a line on keystrokes) when included in the prompt based on the standard file context. The language segments for node, rust tooltips seem to work fine. Removed the go segment for now.

### Theme

Cloud Native Azure - then customised 
[cloud-native-azure.omp.json](https://github.com/JanDeDobbeleer/oh-my-posh/files/13933358/cloud-native-azure.omp.json)


### What OS are you seeing the problem on?

macOS

### Which shell are you using?

zsh

### Log output

```Shell
~  oh-my-posh debug --plain

Version: 19.5.2

Shell: zsh (5.9)

Prompt:

 SPavet01@UK-LC29LHPHVY    ~                                                   zsh   100   22:56:03 
 ~

Segments:

ConsoleTitle(true)                        -   1 ms
Session(true)                             -   8 ms
Path(true)                                -   3 ms
Git(false)                                -   2 ms
Exit(true)                                -   3 ms
Shell(true)                               -   0 ms
Battery(true)                             -  13 ms
Time(true)                                -   0 ms
Text(true)                                -   0 ms

Run duration: 33.500708ms

Cache path: /Users/SPavet01/.cache/oh-my-posh

Config path: /Users/SPavet01/cloud-native-azure.omp.json

Logs:

[DEBUG] 22:56:03.960 shell.go:Getenv:399 → iTerm.app
[TRACE] 22:56:03.960 shell.go:Getenv(TERM_PROGRAM) - 1.08075ms
[DEBUG] 22:56:03.960 shell.go:Getenv:399 → NO DATA
[TRACE] 22:56:03.960 shell.go:Getenv(OMP_CACHE_DIR) - 150.083µs
[DEBUG] 22:56:03.960 shell.go:Getenv:399 → NO DATA
[TRACE] 22:56:03.960 shell.go:Getenv(XDG_CACHE_HOME) - 2.084µs
[TRACE] 22:56:03.960 shell_unix.go:CachePath() - 197.166µs
[DEBUG] 22:56:03.961 shell.go:Getenv:399 → /Users/SPavet01/cloud-native-azure.omp.json
[TRACE] 22:56:03.961 shell.go:Getenv(POSH_THEME) - 3.584µs
[DEBUG] 22:56:03.961 shell_unix.go:Platform:77 → darwin
[TRACE] 22:56:03.961 shell.go:resolveConfigPath() - 169.375µs
[TRACE] 22:56:03.962 shell.go:Init() - 2.547459ms
[TRACE] 22:56:03.962 shell.go:Flags() - 83ns
[TRACE] 22:56:03.963 config.go:loadConfig() - 1.076875ms
[TRACE] 22:56:03.963 shell.go:Flags() - 42ns
[DEBUG] 22:56:03.963 shell.go:Getenv:399 → NO DATA
[TRACE] 22:56:03.963 shell.go:Getenv(OMP_CACHE_DISABLED) - 2.041µs
[DEBUG] 22:56:03.963 shell.go:Shell:662 → no shell name provided in flags, trying to detect it
[DEBUG] 22:56:03.963 shell.go:Shell:670 → process name: zsh
[TRACE] 22:56:03.963 shell.go:Shell() - 401.167µs
[DEBUG] 22:56:03.963 shell.go:Getenv:399 → 5.9
[TRACE] 22:56:03.963 shell.go:Getenv(POSH_SHELL_VERSION) - 2.375µs
[DEBUG] 22:56:03.963 debug.go:PrintDebug:22 → Segment: Title
[DEBUG] 22:56:03.963 text.go:Render:70 → Rendering template: {{ .Shell }} in {{ .Folder }}
[TRACE] 22:56:03.964 shell_unix.go:Root() - 750ns
[TRACE] 22:56:03.964 shell.go:Shell() - 125ns
[TRACE] 22:56:03.964 shell.go:StatusCodes() - 84ns
[DEBUG] 22:56:03.965 shell_unix.go:IsWsl:35 → false
[TRACE] 22:56:03.965 shell_unix.go:IsWsl() - 155.583µs
[DEBUG] 22:56:03.965 shell.go:TemplateCache:843 → environment: [TERM_SESSION_ID=w0t0p0:17DC67CF-F57C-4767-8DDE-D58D54BD6411 SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.8lhej7Kwx3/Listeners LC_TERMINAL_VERSION=3.4.23 COLORFGBG=7;0 ITERM_PROFILE=Default XPC_FLAGS=0x0 LANG=en_GB.UTF-8 PWD=/Users/SPavet01 SHELL=/bin/zsh __CFBundleIdentifier=com.googlecode.iterm2 TERM_PROGRAM_VERSION=3.4.23 TERM_PROGRAM=iTerm.app PATH=/Users/SPavet01/google-cloud-sdk/bin:/Users/SPavet01/.volta/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Library/Apple/usr/bin:/usr/local/share/dotnet:~/.dotnet/tools:/usr/local/go/bin:/Users/SPavet01/.cargo/bin:/Applications/Visual Studio Code.app/Contents/Resources/app/bin:/Users/SPavet01/Library/Application Support/JetBrains/Toolbox/scripts LC_TERMINAL=iTerm2 COLORTERM=truecolor COMMAND_MODE=unix2003 TERM=xterm-256color HOME=/Users/SPavet01 TMPDIR=/var/folders/l1/3h87ptdn79x7db4j9kq8txk40000gq/T/ USER=SPavet01 XPC_SERVICE_NAME=0 LOGNAME=SPavet01 ITERM_SESSION_ID=w0t0p0:17DC67CF-F57C-4767-8DDE-D58D54BD6411 __CF_USER_TEXT_ENCODING=0x1F7:0:2 SHLVL=1 OLDPWD=/Users/SPavet01 HOMEBREW_PREFIX=/opt/homebrew HOMEBREW_CELLAR=/opt/homebrew/Cellar HOMEBREW_REPOSITORY=/opt/homebrew MANPATH=/opt/homebrew/share/man:: INFOPATH=/opt/homebrew/share/info: POSH_THEME=/Users/SPavet01/cloud-native-azure.omp.json POSH_SHELL_VERSION=5.9 POSH_PID=92711 POWERLINE_COMMAND=oh-my-posh CONDA_PROMPT_MODIFIER=false POSH_PROMPT_COUNT=21 CLICOLOR=1 LSCOLORS=gxfxcxdxbxegedabagacad VOLTA_HOME=/Users/SPavet01/.volta _=/opt/homebrew/bin/oh-my-posh]
[TRACE] 22:56:03.965 shell.go:GOOS() - 42ns
[DEBUG] 22:56:03.965 shell.go:Pwd:429 → /Users/SPavet01
[TRACE] 22:56:03.965 shell.go:Pwd() - 22.792µs
[TRACE] 22:56:03.965 shell.go:GOOS() - 41ns
[TRACE] 22:56:03.965 shell.go:GOOS() - 42ns
[TRACE] 22:56:03.965 shell.go:GOOS() - 41ns
[TRACE] 22:56:03.965 shell.go:GOOS() - 42ns
[TRACE] 22:56:03.965 shell.go:GOOS() - 84ns
[TRACE] 22:56:03.965 shell.go:GOOS() - 42ns
[TRACE] 22:56:03.965 shell.go:GOOS() - 41ns
[TRACE] 22:56:03.965 shell.go:GOOS() - 41ns
[TRACE] 22:56:03.965 shell.go:GOOS() - 42ns
[TRACE] 22:56:03.965 shell.go:GOOS() - 42ns
[DEBUG] 22:56:03.965 shell.go:User:564 → SPavet01
[TRACE] 22:56:03.965 shell.go:User() - 1.625µs
[DEBUG] 22:56:03.965 shell.go:Host:576 → UK-LC29LHPHVY
[TRACE] 22:56:03.965 shell.go:Host() - 7.375µs
[TRACE] 22:56:03.965 shell.go:GOOS() - 41ns
[DEBUG] 22:56:03.965 shell.go:Getenv:399 → 1
[TRACE] 22:56:03.965 shell.go:Getenv(SHLVL) - 1.708µs
[TRACE] 22:56:03.965 shell.go:TemplateCache() - 361.083µs
[DEBUG] 22:56:03.965 shell.go:Getenv:399 → NO DATA
[TRACE] 22:56:03.965 shell.go:Getenv(POSH_CURSOR_LINE) - 136.791µs
[DEBUG] 22:56:03.965 shell.go:Getenv:399 → NO DATA
[TRACE] 22:56:03.965 shell.go:Getenv(POSH_CURSOR_COLUMN) - 1.458µs
[TRACE] 22:56:03.965 shell.go:Flags() - 42ns
[TRACE] 22:56:03.965 shell.go:Flags() - 42ns
[TRACE] 22:56:03.965 shell.go:Shell() - 41ns
[TRACE] 22:56:03.965 shell.go:Shell() - 41ns
[TRACE] 22:56:03.966 shell.go:Flags() - 42ns
[TRACE] 22:56:03.966 shell.go:Flags() - 83ns
[TRACE] 22:56:03.966 shell.go:Flags() - 42ns
[TRACE] 22:56:03.966 shell.go:Flags() - 42ns
[TRACE] 22:56:03.966 shell.go:Pwd() - 167ns
[TRACE] 22:56:03.966 shell.go:Pwd() - 375ns
[TRACE] 22:56:03.966 shell.go:Pwd() - 84ns
[TRACE] 22:56:03.966 shell.go:GOOS() - 125ns
[TRACE] 22:56:03.966 shell.go:GOOS() - 42ns
[DEBUG] 22:56:03.966 segment.go:SetEnabled:512 → Segment: Git
[DEBUG] 22:56:03.966 segment.go:SetEnabled:512 → Segment: Path
[DEBUG] 22:56:03.966 segment.go:SetEnabled:512 → Segment: Session
[DEBUG] 22:56:03.966 shell_unix.go:IsWsl:35 → false
[TRACE] 22:56:03.966 shell_unix.go:IsWsl() - 6.208µs
[TRACE] 22:56:03.966 shell_unix.go:IsWsl2() - 8.167µs
[TRACE] 22:56:03.966 shell.go:GOOS() - 42ns
[DEBUG] 22:56:03.966 shell.go:Getenv:399 → NO DATA
[TRACE] 22:56:03.966 shell.go:Getenv(SSH_CONNECTION) - 438.875µs
[DEBUG] 22:56:03.966 shell.go:Getenv:399 → NO DATA
[TRACE] 22:56:03.966 shell.go:Getenv(SSH_CLIENT) - 1.792µs
[DEBUG] 22:56:03.966 shell_unix.go:Platform:77 → darwin
[TRACE] 22:56:03.966 shell.go:Pwd() - 167ns
[TRACE] 22:56:03.966 shell.go:Shell() - 125ns
[TRACE] 22:56:03.966 shell.go:Shell() - 42ns
[DEBUG] 22:56:03.967 properties.go:GetString:28 → {{ .Code }}
[TRACE] 22:56:03.967 shell.go:Pwd() - 125ns
[TRACE] 22:56:03.967 shell.go:GOOS() - 42ns
[DEBUG] 22:56:03.967 segment.go:SetEnabled:512 → Segment: Exit
[TRACE] 22:56:03.967 shell.go:StatusCodes() - 208ns
[DEBUG] 22:56:03.967 text.go:Render:70 → Rendering template: {{ .Code }}
[DEBUG] 22:56:03.967 shell.go:CommandPath:615 → /usr/bin/git
[TRACE] 22:56:03.967 shell.go:CommandPath(git) - 268.458µs
[TRACE] 22:56:03.967 shell.go:HasCommand(git) - 270.459µs
[TRACE] 22:56:03.967 shell.go:Pwd() - 42ns
[ERROR] 22:56:03.968 shell.go:HasParentFilePath:758 → stat .git: no such file or directory
[TRACE] 22:56:03.968 shell.go:HasParentFilePath(.git) - 1.375583ms
[DEBUG] 22:56:03.968 properties.go:GetBool:22 → fetch_bare_info: false
[DEBUG] 22:56:03.967 properties.go:GetBool:22 → mapped_locations_enabled: true
[DEBUG] 22:56:03.968 properties.go:GetString:28 → 
[DEBUG] 22:56:03.968 properties.go:GetString:28 → 
[TRACE] 22:56:03.968 shell.go:GOOS() - 41ns
[DEBUG] 22:56:03.968 properties.go:GetString:28 → ~
[TRACE] 22:56:03.969 shell.go:TemplateCache() - 333ns
[DEBUG] 22:56:03.969 properties.go:GetBool:22 → always_enabled: true
[TRACE] 22:56:03.969 shell.go:TemplateCache() - 83ns
[DEBUG] 22:56:03.969 properties.go:GetKeyValueMap:46 → mapped_locations: map[]
[TRACE] 22:56:03.969 shell.go:PathSeparator() - 83ns
[TRACE] 22:56:03.969 shell.go:GOOS() - 42ns
[TRACE] 22:56:03.969 shell.go:GOOS() - 84ns
[TRACE] 22:56:03.969 shell.go:GOOS() - 42ns
[TRACE] 22:56:03.969 shell.go:GOOS() - 42ns
[TRACE] 22:56:03.969 shell.go:PathSeparator() - 42ns
[TRACE] 22:56:03.969 shell.go:PathSeparator() - 41ns
[TRACE] 22:56:03.969 shell.go:GOOS() - 42ns
[TRACE] 22:56:03.969 shell.go:GOOS() - 0s
[TRACE] 22:56:03.969 shell.go:PathSeparator() - 42ns
[TRACE] 22:56:03.969 shell.go:GOOS() - 42ns
[TRACE] 22:56:03.969 shell.go:GOOS() - 42ns
[TRACE] 22:56:03.969 shell.go:PathSeparator() - 42ns
[TRACE] 22:56:03.969 shell.go:GOOS() - 42ns
[TRACE] 22:56:03.969 shell.go:GOOS() - 42ns
[TRACE] 22:56:03.969 shell.go:PathSeparator() - 83ns
[TRACE] 22:56:03.969 shell.go:Pwd() - 166ns
[DEBUG] 22:56:03.969 shell_unix.go:IsWsl:35 → false
[TRACE] 22:56:03.969 shell_unix.go:IsWsl() - 2.792µs
[TRACE] 22:56:03.969 shell.go:StackCount() - 83ns
[TRACE] 22:56:03.969 shell_unix.go:DirIsWritable(/Users/SPavet01) - 25.125µs
[TRACE] 22:56:03.969 shell.go:TemplateCache() - 5.5µs
[DEBUG] 22:56:03.974 shell.go:RunCommand:594 → SPavet01                      14 Jan 22:56
[TRACE] 22:56:03.974 shell.go:RunCommand(who am i) - 7.802167ms
[TRACE] 22:56:03.974 shell.go:TemplateCache() - 208ns
[DEBUG] 22:56:03.974 text.go:Render:70 → Rendering template:   {{ if gt .Code 0 }} {{ .Meaning }}{{ else }}{{ end }}
[DEBUG] 22:56:03.974 text.go:Render:70 → Rendering template:    {{ .Path }}
[DEBUG] 22:56:03.974 text.go:Render:70 → Rendering template:  {{ if .SSHSession }} {{ end }}{{ .UserName }}@{{ .HostName }}
[TRACE] 22:56:03.975 shell.go:TemplateCache() - 84ns
[TRACE] 22:56:03.975 shell.go:Shell() - 125ns
[TRACE] 22:56:03.975 shell.go:TemplateCache() - 208ns
[TRACE] 22:56:03.975 shell.go:Shell() - 42ns
[TRACE] 22:56:03.975 shell.go:TemplateCache() - 375ns
[TRACE] 22:56:03.975 shell.go:Shell() - 41ns
[DEBUG] 22:56:03.975 text.go:Render:70 → Rendering template: diamond
[DEBUG] 22:56:03.975 text.go:Render:70 → Rendering template: powerline
[DEBUG] 22:56:03.975 text.go:Render:70 → Rendering template: powerline
[DEBUG] 22:56:03.975 text.go:Render:70 → Rendering template: {{ if gt .Code 0 }}#f1184c{{ end }}
[TRACE] 22:56:03.975 shell.go:TemplateCache() - 292ns
[DEBUG] 22:56:03.975 text.go:Render:70 → Rendering template: diamond
[TRACE] 22:56:03.975 shell.go:Shell() - 42ns
[TRACE] 22:56:03.975 shell.go:Shell() - 41ns
[TRACE] 22:56:03.975 shell.go:Shell() - 41ns
[TRACE] 22:56:03.975 shell.go:Shell() - 42ns
[TRACE] 22:56:03.975 shell.go:Flags() - 83ns
[TRACE] 22:56:03.975 shell.go:Flags() - 41ns
[TRACE] 22:56:03.975 shell.go:Flags() - 83ns
[TRACE] 22:56:03.975 shell.go:Pwd() - 125ns
[TRACE] 22:56:03.975 shell.go:Pwd() - 41ns
[TRACE] 22:56:03.975 shell.go:GOOS() - 42ns
[TRACE] 22:56:03.975 shell.go:GOOS() - 0s
[DEBUG] 22:56:03.975 segment.go:SetEnabled:512 → Segment: Time
[DEBUG] 22:56:03.975 segment.go:SetEnabled:512 → Segment: Shell
[DEBUG] 22:56:03.975 properties.go:GetString:28 → 15:04:05
[TRACE] 22:56:03.975 shell.go:TemplateCache() - 42ns
[DEBUG] 22:56:03.975 properties.go:GetKeyValueMap:46 → mapped_shell_names: map[]
[TRACE] 22:56:03.975 shell.go:Shell() - 42ns
[TRACE] 22:56:03.975 shell.go:Flags() - 42ns
[TRACE] 22:56:03.975 shell.go:TemplateCache() - 84ns
[TRACE] 22:56:03.976 shell.go:Pwd() - 83ns
[TRACE] 22:56:03.976 shell.go:GOOS() - 41ns
[DEBUG] 22:56:03.976 segment.go:SetEnabled:512 → Segment: Battery
[DEBUG] 22:56:03.976 shell_unix.go:IsWsl:35 → false
[TRACE] 22:56:03.976 shell_unix.go:IsWsl() - 133.208µs
[DEBUG] 22:56:03.988 shell.go:RunCommand:594 ↓
    Now drawing from 'AC Power'
     -InternalBattery-0 (id=35782755)	100%; charged; 0:00 remaining present: true
[TRACE] 22:56:03.988 shell.go:RunCommand(pmset -g batt) - 12.702375ms
[TRACE] 22:56:03.988 shell_darwin.go:BatteryState() - 12.772166ms
[DEBUG] 22:56:03.988 properties.go:GetString:28 → 
[TRACE] 22:56:03.988 shell.go:TemplateCache() - 5.625µs
[DEBUG] 22:56:03.988 text.go:Render:70 → Rendering template:   {{ .CurrentDate | date .Format }}
[DEBUG] 22:56:03.988 text.go:Render:70 → Rendering template: <#0077c2,transparent></>  {{ .Name }} <transparent,#0077c2></>
[DEBUG] 22:56:03.988 text.go:Render:70 → Rendering template:  {{ if not .Error }}{{ .Icon }}{{ .Percentage }}{{ end }}{{ .Error }}
[TRACE] 22:56:03.989 shell.go:TemplateCache() - 42ns
[TRACE] 22:56:03.989 shell.go:Shell() - 84ns
[TRACE] 22:56:03.989 shell.go:TemplateCache() - 84ns
[TRACE] 22:56:03.989 shell.go:TemplateCache() - 167ns
[TRACE] 22:56:03.989 shell.go:Shell() - 42ns
[TRACE] 22:56:03.989 shell.go:Shell() - 125ns
[DEBUG] 22:56:03.989 text.go:Render:70 → Rendering template: plain
[DEBUG] 22:56:03.989 text.go:Render:70 → Rendering template: {{if eq "Charging" .State.String}}#40c4ff{{end}}
[TRACE] 22:56:03.989 shell.go:TemplateCache() - 83ns
[DEBUG] 22:56:03.989 text.go:Render:70 → Rendering template: {{if eq "Discharging" .State.String}}#ff5722{{end}}
[TRACE] 22:56:03.990 shell.go:TemplateCache() - 42ns
[DEBUG] 22:56:03.990 text.go:Render:70 → Rendering template: {{if eq "Full" .State.String}}#4caf50{{end}}
[TRACE] 22:56:03.990 shell.go:TemplateCache() - 42ns
[DEBUG] 22:56:03.990 text.go:Render:70 → Rendering template: powerline
[DEBUG] 22:56:03.990 text.go:Render:70 → Rendering template: diamond
[TRACE] 22:56:03.990 shell.go:Shell() - 42ns
[DEBUG] 22:56:03.992 shell_unix.go:TerminalWidth:70 → terminal width: 120
[TRACE] 22:56:03.992 shell_unix.go:TerminalWidth() - 2.533625ms
[TRACE] 22:56:03.992 shell.go:Shell() - 42ns
[TRACE] 22:56:03.992 shell.go:Shell() - 83ns
[TRACE] 22:56:03.992 shell.go:Shell() - 42ns
[TRACE] 22:56:03.992 shell.go:Shell() - 42ns
[TRACE] 22:56:03.992 shell.go:Flags() - 83ns
[TRACE] 22:56:03.992 shell.go:Pwd() - 209ns
[TRACE] 22:56:03.992 shell.go:GOOS() - 42ns
[DEBUG] 22:56:03.992 segment.go:SetEnabled:512 → Segment: Text
[TRACE] 22:56:03.992 shell.go:TemplateCache() - 41ns
[DEBUG] 22:56:03.992 text.go:Render:70 → Rendering template:  ~
[TRACE] 22:56:03.992 shell.go:Shell() - 41ns
[DEBUG] 22:56:03.992 shell.go:Getenv:399 → iTerm.app
[TRACE] 22:56:03.992 shell.go:Getenv(TERM_PROGRAM) - 2.375µs
[DEBUG] 22:56:03.992 text.go:Render:70 → Rendering template: plain
[TRACE] 22:56:03.992 shell.go:Shell() - 0s
[TRACE] 22:56:03.992 shell.go:Shell() - 42ns
[TRACE] 22:56:03.992 shell.go:Flags() - 42ns
[TRACE] 22:56:03.992 shell.go:Shell() - 42ns
[TRACE] 22:56:03.992 shell.go:Shell() - 42ns
[TRACE] 22:56:03.992 shell.go:Flags() - 0s
[DEBUG] 22:56:03.992 shell.go:Getenv:399 → NO DATA
[TRACE] 22:56:03.992 shell.go:Getenv(OMP_CACHE_DIR) - 1.75µs
[DEBUG] 22:56:03.992 shell.go:Getenv:399 → NO DATA
[TRACE] 22:56:03.992 shell.go:Getenv(XDG_CACHE_HOME) - 1.209µs
[TRACE] 22:56:03.992 shell_unix.go:CachePath() - 22.042µs
[TRACE] 22:56:03.992 shell.go:Flags() - 42ns
```

--- a/website/docs/segments/languages/golang.mdx
+++ b/website/docs/segments/languages/golang.mdx
@@ -19,7 +19,7 @@ import Config from "@site/src/components/Config.js";
     powerline_symbol: "\uE0B0",
     foreground: "#ffffff",
     background: "#7FD5EA",
-    template: " \u202D\uFCD1 {{ .Full }} ",
+    template: " \ue627 {{ .Full }} ",
   }}
 />
 

